blogArticle.md

```diff
@@ -0,0 +1,590 @@
+
+```javascript
+//app/index.js
+
+import {AsyncStorage} from 'react-native';
+
+class App extends Component {
+
+  constructor(){
+    super();
+    this.state = { hasToken: false };
+  }
+
+  componentWillMount() {
+    AsyncStorage.getItem('id_token').then((token) => {
+      if (token !== null) {
+        this.setState({ hasToken: true });
+      }
+    })
+  }
+
+  render() {
+    return(
+      <Router>
+        <Scene key="root">
+          <Scene
+            component={Authentication}
+            initial={!this.state.hasToken}
+            (...)
+          />
+          <Scene
+            component={HomePage}
+            initial={this.state.hasToken}
+            (...)
+          />
+        </Scene>
+      </Router>
+    )
+  }
+}
+```
+
+There are three reasons why this approach doesn't work:
+1. the access to the `AsyncStorage` is asynchronous, so the `render` method is executed before the state is set
+2. the `componentWillMount` method doesn't trigger a re-rendering if the state changes
+3. even if it component re-rendered, once the `Router` is instantiated, the `initial` property will not be updated
+
+Thus we must find a way to wait for the token's existence verification to finish before returning the `Router` on the `render` method.
+
+To solve this problem, a loader will be returned by default on the `render` method.
+Once the token verification is finished, a 2nd state variable `isLoaded` will tell the render method to return the `Router` with the calculated value for the initial scene:
+
+```javascript
+//app/index.js
+
+import {ActivityIndicator, AsyncStorage} from 'react-native';
+
+class App extends Component {
+
+  constructor(){
+    super();
+    this.state = { hasToken: false, isLoaded: false };
+  }
+
+  componentDidMount() {
+    AsyncStorage.getItem('id_token').then((token) => {
+      if (token !== null){
+        this.setState({
+          hasToken: true,
+          isLoaded: true
+        });
+      } else{
+        this.setState({
+          hasToken: false,
+          isLoaded: true
+        });
+      }
+    });
+  }
+
+  render() {
+    if (!this.state.isLoaded){
+      return (
+        <ActivityIndicator />
+      )
+    } else {
+      return(
+        <Router>
+          <Scene key="root">
+            <Scene
+              component={Authentication}
+              initial={!this.state.hasToken}
+              (...)
+            />
+            <Scene
+              component={HomePage}
+              initial={this.state.hasToken}
+              (...)
+            />
+            </Scene>
+        </Router>
+      )
+    }
+  }
+}
+```
+
+### Conclusion
+
+In this article we've seen how to:
+- share a common codebase to build our *Android* and *iOS* apps;
+- set up routes and scenes with `react-native-router-flux`;
+- communicate to an API to set up a simple JWT authentication system;
+- save and retrieve elements from the `AsyncStorage`;
+- create a persistent user session _*_
+
```

Comment:
Dernier point : je trouve l'article peut-être un peu long. En effet, tu traites beaucoup de choses.
Pourquoi ne pas séparer le début 
* share a common codebase to build our *Android* and *iOS* apps;
 * set up routes and scenes with `react-native-router-flux`;
de l'authentification et de la persistence ?

Tu pourrais faire un article en plusieurs parties.
Quelqu'un qui connait déjà react-native n'a pas envie de lire le début car il connait déjà, mais par contre il peut etre très intéressé par le coeur du sujet.

Tu pourrais commencer l'article en disant:
"Dans mon précédent article (lien) on a vu comment faire une app reactnative ios/andoird avec un navigateur. Vous pouvez trouver le code sur ce repo (lien). On va repartir de ce code pour ajouter l'authentification persistée" ...

Ou quelquechose dans ce genre. Qu'en penses tu ?